amd_openvx_extensions/amd_rpp/source/kernel_rpp.cpp

```diff
@@ -2860,23 +2881,84 @@ void fillAudioDescriptionPtrFromDims(RpptDescPtr &descPtr, size_t *maxTensorDims
     }
 }
 
-void fillGenericDescriptionPtrfromDims(RpptGenericDescPtr &genericDescPtr, vxTensorLayout layout, size_t *maxTensorDims) {
-    if (layout != vxTensorLayout::VX_NHW && layout != vxTensorLayout::VX_NFT && layout != vxTensorLayout::VX_NTF)
-        throw std::runtime_error("Invalid layout value in fillGenericDescriptionPtrfromDims, currently supports only NHW/NFT/NTF layouts");
-    else if(tensorLayoutMapping.find(layout) != tensorLayoutMapping.end())
-        genericDescPtr->layout = tensorLayoutMapping.at(layout);
-
-    genericDescPtr->dims[0] = maxTensorDims[0];
-    genericDescPtr->dims[1] = maxTensorDims[1];
-    genericDescPtr->dims[2] = maxTensorDims[2];
-    genericDescPtr->dims[3] = 1;
-    if(genericDescPtr->dims[2] == 1)
-        genericDescPtr->numDims = 2;
-    else
-        genericDescPtr->numDims = 3;
-    genericDescPtr->strides[0] = genericDescPtr->dims[1] * genericDescPtr->dims[2] * genericDescPtr->dims[3];
-    genericDescPtr->strides[1] = genericDescPtr->dims[2] * genericDescPtr->dims[3];
-    genericDescPtr->strides[2] = genericDescPtr->dims[3];
+void fillGenericDescriptionPtrfromDims(RpptGenericDescPtr &genericDescPtr, vxTensorLayout layout, size_t *tensorDims) {
+    switch(layout) {
+        case vxTensorLayout::VX_NHWC: {
```

Comment:
NCHW, NHWC can be clubbed,
NFHWC and NFCHW can be clubbed

The way we assign strides looks common
Please make it common, instead of adding multiple lines of code

Reply:
Clubbed NCHW and NHWC. Removed cases for NFHWC and NFCHW